src/main/java/org/opentripplanner/routing/algorithm/raptor/transit/TripPatternForDate.java

```diff
@@ -43,25 +44,28 @@
     private final LocalDateTime endOfRunningPeriod;
 
     public TripPatternForDate(
-        TripPatternWithRaptorStopIndexes tripPattern, TripTimes[] tripTimes, LocalDate localDate
+        TripPatternWithRaptorStopIndexes tripPattern,
+        List<TripTimes> tripTimes,
+        LocalDate localDate
     ) {
         this.tripPattern = tripPattern;
-        this.tripTimes = tripTimes;
+        this.tripTimes = new ArrayList<>(tripTimes);
         this.localDate = localDate;
 
         // These depend on the tripTimes array being sorted
         this.startOfRunningPeriod = DateMapper.asDateTime(
             localDate,
-            tripTimes[0].getDepartureTime(0)
+            tripTimes.get(0).getDepartureTime(0)
         );
         this.endOfRunningPeriod = DateMapper.asDateTime(
             localDate,
-            tripTimes[tripTimes.length - 1].getArrivalTime(
-                tripTimes[tripTimes.length - 1].getNumStops() - 1)
+            tripTimes
+                .get(tripTimes.size() - 1)
+                .getArrivalTime(tripTimes.get(tripTimes.size() - 1).getNumStops() - 1)
         );
```

Comment:
I think this is slightly more readable:
```
var last = tripTimes.get(tripTimes.size() - 1);
this.endOfRunningPeriod = DateMapper.asDateTime(
    localDate, 
    last.getArrivalTime(last.getNumStops() - 1)
);    
```

Reply:
I did this small fix, there was a conflict witch I had to resolve locally, so it was easy to do this one at the same time.